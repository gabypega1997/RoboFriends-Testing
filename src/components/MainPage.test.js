import React from "react";

import { shallow, mount } from "enzyme/build";
import MainPage from './MainPage'

let wrapper ,wrapper2, wrapper3;
beforeEach(()=>{
    const mockProps = {
        onRequestRobots: jest.fn(),
        robots:[],
        searchField:'',
        isPending: false
    }
    wrapper = shallow(<MainPage {...mockProps}/>)
})


describe('App container',()=>{
    it('expect to render App component', () => {
        expect(wrapper).toMatchSnapshot();
    }),
    it('filters robots correctly', () => {
        const mockProps = {
            onRequestRobots: jest.fn(),
            robots:[{
                id:3,
                name:'John',
                email:'john@gmail.com'
            }],
            searchField:'john',
            isPending: false
        }
        wrapper2 = shallow(<MainPage {...mockProps}/>)
        expect(wrapper.instance().filteredRobots([])).toEqual([])
        expect(wrapper2.instance().filteredRobots([])).toEqual([
            {
                id:3,
                name:'John',
                email:'john@gmail.com'
            }
        ])
    })
})

describe('Is Pending is True', () => { 
    const mockProps = {
        onRequestRobots: jest.fn(),
        robots:[{
            id:3,
            name:'John',
            email:'john@gmail.com'
        }],
        searchField:'john',
        isPending: true
    }
    wrapper3 = shallow(<MainPage {...mockProps}/>)
    expect(wrapper3.instance().render()).toEqual(<h1>Loading</h1>);
})


