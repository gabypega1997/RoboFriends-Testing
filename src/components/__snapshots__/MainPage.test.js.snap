// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App container expect to render App component 1`] = `
<div
  className="tc"
>
  <h1
    className="f1"
  >
    RoboFriends
  </h1>
  <SearchBox />
  <Scroll>
    <CardList
      robots={Array []}
    />
  </Scroll>
</div>
`;
